State.cpp

```diff
@@ -0,0 +1,248 @@
+#include "BreathData.h"
+#include "State.h"
+#include "Constants.h"
+#include "Display.h"
+#include "Flow.h"
+#include "Pressure.h"
+#include "Valve.h"
+#include "ProportionalValve.h"
+#include "AlarmManager.h"
+#include "State.h"
+
+// Initialize singleton instances
+OffState* OffState::instance = 0;
+InsStateVC* InsStateVC::instance = 0;
+InsHoldStateVC* InsHoldStateVC::instance = 0;
+ExpStateVC* ExpStateVC::instance = 0;
+PeepStateVC* PeepStateVC::instance = 0;
+ExpHoldStateVC* ExpHoldStateVC::instance = 0;
+
+BreathData* State::breath;
+
+/*********************************************************/
+/* OffState                                              */
+/*********************************************************/
+
+// performs actions necessary upon entering a state
+static State* OffState::begin(BreathData b) {
+    if (instance==0) instance = new OffState();  // one-time setup
```

Comment:
There is no need to use pointers and allocate from the heap, nor do you need an `if` statement. Simply write:
```
  OffState instance;
...
  return &instance;
```
The compiler will do the rest more efficiently than you can.

Reply:
I copied this from examples of the singleton design pattern in C++.  If we write it the way you suggest, won't it create a new instance each time we call `enter` or `begin`?